File: lib/models/post.dart
class Post {
  final String id;
  final String imageUrl;
  final String username;
  final String caption;
  final int likes;
  final String? filter;

  Post({
    required this.id,
    required this.imageUrl,
    required this.username,
    required this.caption,
    required this.likes,
    this.filter,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'].toString(),
      imageUrl: json['imageUrl'],
      username: json['username'],
      caption: json['caption'] ?? '',
      likes: json['likes'] ?? 0,
      filter: json['filter'],
    );
  }
}

File: vsco_clone_backend/src/main/java/com/vsco/clone/model/Post.java
package com.vsco.clone.model;

import jakarta.persistence.*;

@Entity
@Table(name = "posts")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT")
    private String imageUrl;
    
    private String username;
    private String caption;
    private int likes;
    private String filter;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }
    public String getCaption() { return caption; }
    public void setCaption(String caption) { this.caption = caption; }
    public int getLikes() { return likes; }
    public void setLikes(int likes) { this.likes = likes; }
    public String getFilter() { return filter; }
    public void setFilter(String filter) { this.filter = filter; }
}
